### PR TITLE
dynamic fuckups

### PR DESCRIPTION
**Pull request by whateverusername0**
## About the PR

fixed midround rolls keeping on rolling after losing all budget
increased the base delay of midround rolls to 20-30 minutes
and did an overall cleanup of unnecessary shit that i left there

i still don't know what the fuck causes enemy communication intercept message to go through 3 times in a row it works just fine on localhost :finnadie: can't debug shit in detroit
image

no cl no fun

## Summary by CodeRabbit

- **New Features**
	- Enhanced logging for midround event failures due to budget constraints.
	- Adjusted event scheduling logic for antag events with new delay parameters.

- **Improvements**
	- Streamlined threat level calculations in the dynamic rule system.
	- Improved clarity and functionality in event rerolling logic.

- **Bug Fixes**
	- Corrected handling of reroll attempts for spawning midround events. 

- **Documentation**
	- Updated comments for consistency across various components.

### CODE REVIEW

**Comment by whateverusername0** [comment]
@coderabbitai do not review

File: Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs [hunk 1]
```diff
@@ -117,7 +117,7 @@ protected override void Started(EntityUid uid, DynamicRuleComponent component, G
         var maxThreat = players.Count < lowpopThreshold ? lowpopThreat : component.MaxThreat;
 
         // generate a random amount of points
-        component.ThreatLevel = LorentzToAmount(component.ThreatCurveCentre, component.ThreatCurveWidth, maxThreat);
+        component.ThreatLevel = LorentzToAmount(0, 1.8f, maxThreat);
```
Comment (Piras314) [inline Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs:120]: I love hardcoded numbers

File: Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs [hunk 2]
```diff
@@ -150,7 +150,6 @@ protected override void Started(EntityUid uid, DynamicRuleComponent component, G
             var ruleset = WeightedPickRule(draftedRules);
 
             if (ruleset == null)
-                // todo write something debug related here
```
Comment (Piras314) [inline Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs:153]: whyd you remove this

File: Content.Server/_Goobstation/StationEvents/DynamicStationEventSchedulerComponent.cs [hunk 3]
```diff
@@ -9,29 +9,18 @@ namespace Content.Server.StationEvents;
...
-    ///     The first midround antag roll will happen 20 minutes into the shift.
-    ///     After that it's all about random.
-    /// </summary>
-    [DataField] public float FirstEventDelay = 1200f;
-
-    /// <summary>
-    ///     
-    /// </summary>
-    [DataField] public float ThreatPerMidroundRoll = 7f;
+    [DataField] public MinMax Delays = new(20 * 60, 30 * 60);
```
Comment (Piras314) [inline Content.Server/_Goobstation/StationEvents/DynamicStationEventSchedulerComponent.cs:25]: hardcode :godmode:

File: Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs (hunk 2 above)
Comment (whateverusername0) [reply Content.Server/_Goobstation/GameTicking/Rules/DynamicRuleSystem.cs:153]: usiless

File: Content.Server/_Goobstation/StationEvents/DynamicStationEventSchedulerComponent.cs (hunk 3 above)
Comment (whateverusername0) [reply Content.Server/_Goobstation/StationEvents/DynamicStationEventSchedulerComponent.cs:25]: do you seriously expect me to write that in yaml :godmode:

**Pull request merged**